nix/module.nix

```diff
@@ -0,0 +1,283 @@
+        description = ''
+          The Telegram username of the account associated with this Seance instance.
+          Commands -- and text to proxy -- will only be accepted from this user.
+        '';
+      };
+
+      secureConfigFile = mkOption {
+        type = types.path;
+        description = ''
+          A file containing any strings to be passed as a configuration file.
+          Should at least contain a `token = <telegram-token>` line.
+
+          Example contents might be:
+
+          [Telegram]
+          token = mySecureTokenHere
+        '';
+      };
+
+      pattern = mkOption {
+        type = types.str;
+        description = ''
+          A regular expression matching the pattern that should be used as a "proxy tag".
+          Text that matches the "content" capture pattern will be proxied as the target uset.
+
+          As an example "^(?:([nN ][:#;.])|([n;#.][nN]))(?P<content>.*)$" matches messages
+          matching patterns such as n#, n:, N;, #n, #N, and etc.
+        '';
+      };
+
+      environment = mkOption {
+        type = types.listOf types.str;
+        default = [];
+        defaultText = "[]";
+        description = ''
+          Any additional variables that should be set in the environment; in the same syntax
+          as when specifying systemd services; i.e. NAME=value.
+        '';
+      };
+
+    };
+  };
+
+  #
+  # Function that translates a discordUser into an equivalent systemd.service entry.
+  #
+  serviceFromTelegramUser = name: user: {
+
+    # The raw systemd unit for each user.
+    "seance-telegram-${name}" = {
+      description = "Seance Telegram Bot for ${name}";
+
+      # Start automatically once our network is up.
+      wantedBy = [ "multi-user.target" ];
+      wants    = [ "network-online.target" ];
+      after    = [ "network-online.target"  ];
+
+      serviceConfig = {
+        Type = "notify";
+        ExecStart = "${cfg.package}/bin/seance-telegram --config %d/configFile --systemd-notify";
+
+        # Pass in the secure environment file, which sets the user's key.
+        LoadCredential = "configFile:${user.secureConfigFile}";
+
+        # Pass in the remainder of the options.
+        Environment = [
+          "PYTHONUNBUFFERED=1"
+          "SEANCE_TELEGRAM_REF_USERNAME=\"${toString user.referenceUsername}\""
+          "SEANCE_TELEGRAM_PATTERN=\"${user.pattern}\""
+        ] ++ user.environment;
+
+        # If this crahses, try again after 5s.
+        Restart = "on-failure";
+        RestartSec = "5s";
+
+        # Run this as a dynamic, non-root user.
+        DynamicUser = true;
+      };
+    };
+  };
+
+in {
+  options = {
+    services.seance = {
+
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Enables Seance servers for each specified user.
+        '';
+      };
+
+      package = mkOption {
+        type = types.path;
+        default = self.packages.${pkgs.system}.default;

```

Comment:
Is there a reason to take all of `self`, which is only used here, instead of just taking the package in the first place?